### PR TITLE
swarm: fix uptime gauge update goroutine leak

### PR DESCRIPTION
**Pull request by justelad**
This PR addresses a goroutine leak in test environment that would appear in the edge case where multiple tests would call the same service' Start method.

fixes https://github.com/ethersphere/go-ethereum/issues/1156 by introducing cleanup functions to allow for custom shutdown callbacks.

### CODE REVIEW

File: swarm/swarm.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ type Swarm struct {
 	swap              *swap.Swap
 	stateStore        *state.DBStore
 	accountingMetrics *protocols.AccountingMetrics
-	startTime         time.Time
+	cleanupFuncs      []func() error
```
Comment (zelig) [inline swarm/swarm.go]: why is this a slice?

Comment (justelad) [reply swarm/swarm.go]: To allow multiple cleanup funcs?
